shard/shard.go

```diff
@@ -20,10 +20,24 @@
 
 package shard
 
+const (
+	// INITIALIZING represents a shard newly assigned to an instance
+	INITIALIZING State = iota
```

Comment:
Let's stick with our current const naming scheme, this should just be `Initializing`.